src/opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp

```diff
@@ -291,102 +291,102 @@ namespace Opm {
+            supportedDoubleKeywords.emplace_back( kw, ISOWCRLookup, "1", true );
 
         // keywords to specify the scaled critical oil-in-gas saturation.
         for( const auto& kw : { "SOGCR", "SOGCRX", "SOGCRX-", "SOGCRY", "SOGCRY-", "SOGCRZ", "SOGCRZ-" } )
-            supportedDoubleKeywords.emplace_back( kw, SOGCRLookup, "1" );
+            supportedDoubleKeywords.emplace_back( kw, SOGCRLookup, "1", true );
         for( const auto& kw : { "ISOGCR", "ISOGCRX", "ISOGCRX-", "ISOGCRY", "ISOGCRY-", "ISOGCRZ", "ISOGCRZ-" } )
-            supportedDoubleKeywords.emplace_back( kw, ISOGCRLookup, "1" );
+            supportedDoubleKeywords.emplace_back( kw, ISOGCRLookup, "1", true );
 
         // keywords to specify the scaled critical water saturation.
         for( const auto& kw : { "SWCR", "SWCRX", "SWCRX-", "SWCRY", "SWCRY-", "SWCRZ", "SWCRZ-" } )
-            supportedDoubleKeywords.emplace_back( kw, SWCRLookup, "1" );
+            supportedDoubleKeywords.emplace_back( kw, SWCRLookup, "1", true );
         for( const auto& kw : { "ISWCR", "ISWCRX", "ISWCRX-", "ISWCRY", "ISWCRY-", "ISWCRZ", "ISWCRZ-" } )
-            supportedDoubleKeywords.emplace_back( kw, ISWCRLookup, "1" );
+            supportedDoubleKeywords.emplace_back( kw, ISWCRLookup, "1", true );
 
         // keywords to specify the scaled oil-water capillary pressure
         for( const auto& kw : { "PCW", "PCWX", "PCWX-", "PCWY", "PCWY-", "PCWZ", "PCWZ-" } )
-            supportedDoubleKeywords.emplace_back( kw, PCWLookup, "1" );
+            supportedDoubleKeywords.emplace_back( kw, PCWLookup, "1", true );
         for( const auto& kw : { "IPCW", "IPCWX", "IPCWX-", "IPCWY", "IPCWY-", "IPCWZ", "IPCWZ-" } )
-            supportedDoubleKeywords.emplace_back( kw, IPCWLookup, "1" );
+            supportedDoubleKeywords.emplace_back( kw, IPCWLookup, "1", true );
 
         // keywords to specify the scaled gas-oil capillary pressure
         for( const auto& kw : { "PCG", "PCGX", "PCGX-", "PCGY", "PCGY-", "PCGZ", "PCGZ-" } )
-            supportedDoubleKeywords.emplace_back( kw, PCGLookup, "1" );
+            supportedDoubleKeywords.emplace_back( kw, PCGLookup, "1", true );
         for( const auto& kw : { "IPCG", "IPCGX", "IPCGX-", "IPCGY", "IPCGY-", "IPCGZ", "IPCGZ-" } )
-            supportedDoubleKeywords.emplace_back( kw, IPCGLookup, "1" );
+            supportedDoubleKeywords.emplace_back( kw, IPCGLookup, "1", true );
 
         // keywords to specify the scaled water relative permeability
         for( const auto& kw : { "KRW", "KRWX", "KRWX-", "KRWY", "KRWY-", "KRWZ", "KRWZ-" } )
-            supportedDoubleKeywords.emplace_back( kw, KRWLookup, "1" );
+            supportedDoubleKeywords.emplace_back( kw, KRWLookup, "1", true );
         for( const auto& kw : { "IKRW", "IKRWX", "IKRWX-", "IKRWY", "IKRWY-", "IKRWZ", "IKRWZ-" } )
-            supportedDoubleKeywords.emplace_back( kw, IKRWLookup, "1" );
```

Comment:
This code is not for the faint of heart ....

Reply:
> This code is not for the faint of heart ....

Agreed.  I tried to not rock the boat too animatedly.